### DISCUSSION TITLE
Meetups API Call Change

### DISCUSSION THREAD

**Discussion by mtranter** [newwwie/newwwie.com#103] [Polls]
**Problem**
The current approach to gathering the Meetups.com data requires looping through a list of known meetup names and making a graphql call for each, and then doing some fairly hideous munging to pull everything together for display. (I wrote this. Sorry.)

**Solution**
We can replace this approach with a call to the meetup.com keywordSearch graphql field and get everything we need in one call.

**Theres No Free Lunch**
With this new method there is no explicit way of supplying a GROUP filter. So In my demo below, I've used a Geo filter (lat/long of Newcastle) and a topicCategoryid filter (technology topicCategoryid).
Currently this means we would see an extra meetup on our Newwwie page (https://www.meetup.com/newcastle-seo-beers/events/299326447) that is not currently there. You can test the query here: https://www.meetup.com/api/playground/#graphQl-playground

So the question is: do we want explicit control of the meetups or are we happy to show all Newcastle based Technology meetups?

```gql
query {
  keywordSearch(
    filter: {
      query: "%"
      lat: -32.9283
      lon: 151.7817
      radius: 20
      source: EVENTS
      topicCategoryId: 546
    }
    sort: { sortField: DATETIME, sortOrder: DESC }
  ) {
    edges {
      node {
        result {
          ...on Event {
            title
            description
            dateTime
            eventUrl
            going
            maxTickets
            duration
            imageUrl
            venue {
              name
              lat
              lng
              address
              city
            }
            group {
              name
              urlname
              groupPhoto {
                id
                baseUrl
                preview
              }
              logo {
                id
                baseUrl
                preview
              }
            }
          }
        }
      }
    }
  }
}
```

**Comment by neozenith** [comment]
I am not a fan of the geographic filter. There are meetup groups in Newcastle that are not friendly affiliated with the Newwwie group / entity. So a fixed allow list seems more appropriate.

That is just my opinion.

**Comment by clbaita** [comment]
I agree with @neozenith here, having a list makes sense and we can guarantee accuracy. Albeit fetching may be able to be simplified.

**Comment by mtranter** [comment]
I like the reasoning @neozenith, not so much re: the affiliation, but the "not friendly" bit could be an issue. 👍🏼
